core/src/main/java/com/radixdlt/monitoring/MetricInstaller.java

```diff
@@ -90,10 +94,12 @@ public final class MetricInstaller {
   @Inject
   public MetricInstaller(
       final SelfValidatorInfo self,
+      final REv2LedgerInitializerToken ledgerInitialization,
       final InMemorySystemInfo inMemorySystemInfo,
       final PeersView peersView,
       final TimeSupplier timeSupplier) {
     this.self = self;
+    this.ledgerInitialization = ledgerInitialization;
```

Comment:
I wonder if this means we initialize metrics later than we used to? Maybe this isn't a problem? or previously it was kinda random if it initialized before genesis executed or not?

Or maybe the prometheus server didn't start until after genesis anyway?

Reply:
> Or maybe the prometheus server didn't start until after genesis anyway?

hah, the prometheus server is started literally as a last-but-one thing that happens: https://github.com/radixdlt/babylon-node/blob/5c88d9489af5527dd1684f97f309e1e11ac81be1/core/src/main/java/com/radixdlt/RunningRadixNode.java#L136

So yeah, the moment at which the (trivial) metrics' constructor is run does not matter here.

Reply:
Haha love it.

Although, just to point out, we somehow reliably observe the `BOOTING` state if something crashes between bootup and first DB read / commit (eg with the failing foundation validator last week) - but that would possibly also be true under this PR I imagine :+1: or maybe not, but the Babylon node corrupts less, so it's not a problem